src/clojure_lsp/refactor/transform.clj

```diff
@@ -109,6 +109,13 @@
        (not (contains? thread-invalid-symbols
                        (some-> zloc z/next z/sexpr)))))
 
+(defn can-thread-at-cursor? [zloc]
+  (or (can-thread? zloc)
+      (and (= (z/tag zloc) :token)
+           (= (z/tag (z/up zloc)) :list)
+           (not (contains? thread-invalid-symbols
+                           (some-> zloc z/up z/next z/sexpr))))))
+
 (defn thread-first
```

Comment:
Should we change all other references that use `can-thread?` to use the new `can-thread-at-cursor?`

Reply:
Good catch. I've renamed the functions to be more clear: `can-thread?` checks both if the current zloc is a list or if it's a token within a list, and `can-thread-list?` only checks if the current zloc is a list. We want only the latter when calling `thread-all` to prevent constantly rechecking the updated zloc, as `(z/up next-loc)` points to the in-progress thread macro rewrite, which will always start with `->` or `->>`.